tests/wps_restapi/test_providers.py

```diff
@@ -14,49 +14,75 @@
     setup_mongodb_processstore,
     setup_mongodb_servicestore
 )
+from weaver.config import WEAVER_CONFIGURATION_ADES, WEAVER_CONFIGURATION_HYBRID
+from weaver.datatype import Service
 from weaver.execute import EXECUTE_CONTROL_OPTION_ASYNC, EXECUTE_TRANSMISSION_MODE_REFERENCE
 from weaver.formats import CONTENT_TYPE_APP_JSON, CONTENT_TYPE_APP_NETCDF, CONTENT_TYPE_APP_ZIP, CONTENT_TYPE_TEXT_PLAIN
 from weaver.utils import fully_qualified_name
 
 
 # pylint: disable=C0103,invalid-name
-class WpsRestApiProcessesTest(unittest.TestCase):
-    remote_server = None
+class WpsProviderBase(unittest.TestCase):
+    remote_provider_name = None
+    settings = {}
+
+    def fully_qualified_test_process_name(self):
+        return fully_qualified_name(self).replace(".", "-")
+
+    def register_provider(self, clear=True, error=False):
+        if clear:
+            self.service_store.clear_services()
+        path = "/providers"
+        data = {"id": self.remote_provider_name, "url": resources.TEST_REMOTE_SERVER_URL}
+        resp = self.app.post_json(path, params=data, headers=self.json_headers, expect_errors=error)
+        if error:
+            assert resp.status_code != 201, "Expected provider to fail registration, but erroneously succeeded."
+        else:
+            err = resp.json
+            assert resp.status_code == 201, "Expected failed provider registration to succeed. Error:\n{}".format(err)
```

Comment:
Not sure I understand why we expect a failed provider registration to succeed.

Reply:
The expected success (201) did not occur, hence we got failed provider registration.